subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/TestRetryPluginSmokeTest.groovy

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Issue
+
+class TestRetryPluginSmokeTest extends AbstractSmokeTest {
+
+    @Issue('https://plugins.gradle.org/plugin/org.gradle.test-retry')
+    def 'test retry plugin'() {
+        when:
+        sourceFile()
+        testSourceFile()
+        buildFile << """
+            plugins {
+                id "java"
+                id "org.gradle.test-retry" version "${TestedVersions.testRetryPlugin}"
+            }
+
+            ${jcenterRepository()}
+
+            dependencies {
+                testImplementation("org.junit.jupiter:junit-jupiter-api:5.5.2")
+                testImplementation("org.junit.jupiter:junit-jupiter-params:5.5.2")
+                testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.5.2")
+            }
+
+            test {
+                doFirst {
+                    file("marker.file").delete()
+                }
+            
+                useJUnitPlatform()
+                retry {
+                    maxRetries = 2
+                }
+            }"""
+
+        then:
+        runner('test').build()
```

Comment:
+1 Stefan's comments about checking that this executed the tests the way we expected them to.

Reply:
I'll add some basic verification of the expected outcome / test output. I think for a smoke test this is enough. We test all supported frameworks in detail in a plugin specific ci nightly job running against the lastest gradle nightly: https://builds.gradle.org/buildConfiguration/GradlePlugins_GradleTestRetryPlugin_null_CrossVersionTestGradleNightliesLinuxJava18

Reply:
done